text_generation/causal_lm/cpp/continuous_batching/library/src/sequence_group.hpp

```diff
@@ -384,4 +410,58 @@ class SequenceGroup {
         }
         return false;
     }
+    
+    GenerationStream::Ptr get_generation_stream() {
+        return m_generation_stream;
+    }
+
+    void finish_generation_stream(GenerationResultStatus status) {
+        m_generation_stream->finish_generation_stream(status);
+    }
+
+    bool handle_dropped() {
+        return m_generation_stream->handle_dropped();
+    }
+
+    void notify_handle() {
+        GenerationOutputs outputs;
+
+        // For beam search streaming is not available, so we notify only upon finishing
+        if(m_sampling_params.is_beam_search()) {
+            if (has_finished()) {
+                std::vector<Sequence::CPtr> finished_sequences = get_finished_sequences();
+
+                OPENVINO_ASSERT(finished_sequences.size() == num_total_seqs() && has_finished());
+                for (auto& sequence: finished_sequences) {
+                    GenerationOutput output;
+                    output.generated_token_ids = sequence->get_generated_ids();
+                    output.score = sequence->get_beam_search_score(m_sampling_params);
+                    outputs.emplace(sequence->get_grouped_id(), output);
+                }
+
+                if (outputs.size()) {
+                    m_generation_stream->push(outputs);
+                }
+            }
+        // For greedy or multinomial sampling we decide whever to stream partial results depending on the user parameter
+        } else if (m_sampling_params.is_greedy_sampling() || m_sampling_params.is_multinomial()) {
+            // TO DO: Now we always stream for greedy search for the sake of benchmarking
+            if (true /* m_sampling_params.stream */) {
+                // If sequence group has been preempted we skip retriving the results as seqeunce tokens have not been cleared and
+                // there's no new token. We reset preemption flag, so when it's scheduled again we process it normally.
+                if (m_preempted) {
+                    m_preempted = false;
+                    return;
+                }
+                for (auto& sequence : m_sequences) {
+                    if (sequence->get_generated_len() > 0) {
+                        outputs.emplace(sequence->get_grouped_id(), sequence->get_last_generation_output(m_sampling_params));
+                    }
+                }
+                if (outputs.size()) {
```

Comment:
why it can be empty?

Reply:
Streaming case with "empty" notification. This prevents us from sending data to the handle if step didn't produce anything (because it's still in the prompt processing phase)